src/main/java/com/apptopia/emr/javaagent/AddRequireToClinitClassVisitor.java

```diff
@@ -0,0 +1,69 @@
+package com.apptopia.emr.javaagent;
+
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+
+import java.net.URL;
+
+import static org.objectweb.asm.Opcodes.ACC_STATIC;
+import static org.objectweb.asm.Opcodes.RETURN;
+
+public final class AddRequireToClinitClassVisitor extends ClassVisitor {
+    private final ClassLoader loader;
+
+    private String className;
+    private int version;
+    private boolean skip;
+
+    private boolean visitedStaticBlock = false;
+
+    AddRequireToClinitClassVisitor(int api, ClassVisitor visitor, ClassLoader loader) {
```

Comment:
I think this constructor should be `public`. Or not?

Reply:
This class should not be used by anyone except for the Agent itself. I'd rather make the class package-private?

Reply:
Well it's a package private already. But usually package private constructors looks weird especially when this is your library and any project person can change the access. There is also some special cases with inheritance.

Reply:
It's public now